src/main/java/nextstep/subway/domain/object/Sections.java

```diff
@@ -0,0 +1,102 @@
+package nextstep.subway.domain.object;
+
+import nextstep.subway.domain.Section;
+import nextstep.subway.domain.Station;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Embeddable
+public class Sections {
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
+    private List<Section> values;
+
+    public Sections() {
+        this.values = new ArrayList<>();
+    }
+
+    public void add(Section section) {
+        if (!validateAddSection(section)) {
+            throw new InvalidParameterException();
+        }
+        this.values.add(section);
+    }
+
+    public Section last() {
+        if (this.values.isEmpty()) {
+            return null;
+        }
+        return this.values.get(this.values.size() - 1);
+    }
+
+    public Station lastDownStation() {
+        return last().getDownStation();
+    }
+
+    public Long lastDownStationId() {
+        return last().getDownStationId();
+    }
+
+    public boolean equalsLastDownStation(Station upStation) {
+        return this.lastDownStation().equals(upStation);
+    }
+
+    public boolean checkDuplicatedDownStation(Station downStation) {
+        return this.values.stream()
+                .anyMatch(section ->
+                        section.getUpStation().equals(downStation) ||
+                                section.getDownStation().equals(downStation)
+                );
+    }
+
+    public List<Station> getAllStations() {
+        List<Station> stations = this.values.stream()
+                .sorted(Comparator.comparing(Section::getId))
+                .map(Section::getUpStation)
+                .collect(Collectors.toList());
+        stations.add(lastDownStation());
+        return stations;
+    }
+
+    public int size() {
+        return this.values.size();
+    }
+
+    public boolean validateAddSection(Section section) {
+        if (values.isEmpty()) {
+            return true;
+        }
+
+        if (!equalsLastDownStation(section.getUpStation())) {
+            return false;
+        }
+
+        return !checkDuplicatedDownStation(section.getDownStation());
```

Comment:
```suggestion
        return values.isEmpty() 
                || equalsLastDownStation(section.getUpStation())
                || !checkDuplicatedDownStation(section.getDownStation());
```
이정도로 정리할 수 있을 것 같아요 😄